lib/containers/ConnectionParams.jsx

```diff
@@ -0,0 +1,142 @@
+ * DISCLAIMED. IN NO EVENT SHALL NORDIC SEMICONDUCTOR ASA OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR
+ * TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF
+ * THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/* eslint react/forbid-prop-types: off */
+/* eslint react/require-default-props: off */
+
+'use strict';
+
+import PropTypes from 'prop-types';
+import React from 'react';
+import Button from 'react-bootstrap/Button';
+import Modal from 'react-bootstrap/Modal';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+
+import * as AdapterActions from '../actions/adapterActions';
+import ConnectionParamsControl from '../components/ConnectionParamsControl';
+
+class ConnectionParams extends React.PureComponent {
+    state = {
+        connectionParameters: null,
+    };
+
+    handleParamChange = connectionParameters =>
+        this.setState({ connectionParameters });
+
+    handleApply() {
+        const { hideConnectionParamDialog, setConnectionParams } = this.props;
+        const { connectionParameters } = this.state;
+        if (!connectionParameters) {
+            return;
+        }
+        setConnectionParams(connectionParameters);
+        hideConnectionParamDialog();
+    }
+
+    handleCancel() {
+        const { hideConnectionParamDialog } = this.props;
+        hideConnectionParamDialog();
+    }
+
+    render() {
+        const { showConnectionParams, connectionParameters } = this.props;
+
+        return (
+            <Modal
+                className="adv-param-modal"
+                show={showConnectionParams}
+                onHide={() => {
+                    this.handleCancel();
+                }}
+            >
+                <Modal.Header>
+                    <Modal.Title>Connection parameters</Modal.Title>
+                </Modal.Header>
+                <Modal.Body>
+                    <ConnectionParamsControl
+                        onChange={this.handleParamChange}
+                        connectionParameters={connectionParameters}
+                    />
+                </Modal.Body>
+                <Modal.Footer>
+                    <div className="form-group">
+                        <Button
+                            type="button"
+                            onClick={() => this.handleApply()}
+                            className="btn btn-primary btn-sm btn-nordic"
+                        >
+                            Apply
+                        </Button>
+                        <Button
+                            type="button"
+                            onClick={() => this.handleCancel()}
```

Comment:
This can be shortened to `onClick={this.handleCancel}`

Reply:
Same goes for the `onHide` and `onClick` handlers above.

Reply:
Actually `handleCancel` is just calling `hideConnectionParamDialog`, so you could just remove that and call `this.props.hideConnectionParamDialog` directly.

E.g.
```js
onClick={this.props.hideConnectionParamDialog}
```